HtmlDiff/Diff.cs

```diff
@@ -197,11 +240,55 @@ private void ProcessDeleteOperation(Operation operation, string cssClass)
             InsertTag("del", cssClass, text);
         }
 
-        private void ProcessEqualOperation(Operation operation)
+        private void ProcessEqualOperation(Operation operation, bool changesAfter)
         {
             string[] result =
                 _newWords.Where((s, pos) => pos >= operation.StartInNew && pos < operation.EndInNew).ToArray();
-            _content.Append(String.Join("", result));
+            if (LinesContext > 0)
+            {
+                int linesInSection = result.Count(m => m.StartsWith("\r\n"));
+                if (linesInSection > LinesContext * 2) // Only put breaks in if the gap is bigger than the size of the content
+                {
+                    // Need to split the context down.
+                    int lines=0;
+                    foreach (var item in result)
+                    {
+                        if (lines < LinesContext || lines > linesInSection-LinesContext)
+                            _content.Append(item);
+                        if (item.StartsWith("\r\n"))
+                            lines++;
+
+                        // Close the previous context
+                        if (lines == LinesContext && _startSectionAdded)
+                        {
+                            _startSectionAdded = false;
+                            _content.Append("</div> <!-- end diff section -->\r\n\r\n");
+                            if (!changesAfter)
+                                return;
+                        }
+
+                        // Now start the new context
+                        if (lines == linesInSection - LinesContext && !_startSectionAdded)
+                        {
+                            _startSectionAdded = true;
+                            _content.AppendFormat("<div class=\"{0}\">  <!-- start diff section -->\r\n", ContextSectionClass);
+                        }
+                    }
+                }
+                else
+                {
+                    if (!_startSectionAdded)
+                    {
+                        _startSectionAdded = true;
+                        _content.AppendFormat("<div class=\"{0}\">  <!-- start diff section -->\r\n", ContextSectionClass);
```

Comment:
This is duplicated a few times. Perhaps introduce a static method to return the template for the start/end of sections. Even better would be to allow callers to specify their own template mechanism. i.e. Introduce an IDiffSummaryTemplate with a default implementation as per the original code above.